config.m4

```diff
@@ -111,6 +111,12 @@ if test "$MONGODB" != "no"; then
     PHP_CHECK_GCC_ARG(-Wdeclaration-after-statement,    _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wdeclaration-after-statement")
     dnl PHP_CHECK_GCC_ARG(-Werror,                          _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Werror")
 
+    PHP_MONGODB_FOUND_VERSION=`${PHP_CONFIG} --version`
+    PHP_MONGODB_FOUND_VERNUM=`echo "${PHP_MONGODB_FOUND_VERSION}" | $AWK 'BEGIN { FS = "."; } { printf "%d", ([$]1 * 100 + [$]2) * 100 + [$]3;}'`
+
+    if test "$PHP_MONGODB_FOUND_VERNUM" -ge "50400"; then
+      PHP_CHECK_GCC_ARG(-Wno-cast-qual,                 _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wno-cast-qual")
+    fi
```

Comment:
I'm fine with this.
Just for reference, which exact case prompted this change?
Isn't it also valid for the arginfos?